ln-gateway/src/bin/ln_gateway.rs

```diff
@@ -1,242 +1,83 @@
-
-    // Write gateway config
-    let gateway_cfg = GatewayClientConfig {
-        client_config: client_cfg.clone(),
-        redeem_key: kp_fed,
-        timelock_delta: 10,
-        node_pub_key,
-        api: Url::parse(format!("http://{}", bind_addr).as_str())
-            .expect("Could not parse URL to generate GatewayClientConfig API endpoint"),
-    };
-    let gw_cfg_file_path: PathBuf = workdir.join("gateway.json");
-    let gw_cfg_file = std::fs::File::create(gw_cfg_file_path).expect("Could not create cfg file");
-    serde_json::to_writer_pretty(gw_cfg_file, &gateway_cfg).unwrap();
-}
+        .expect("Failed to register cln rpc");
 
-/// Loads configs if they exist, generates them if not
-/// Initializes [`LnGateway`] and runs it's main event loop
-async fn initialize_gateway(
-    plugin: &Plugin<PluginState>,
-    sender: mpsc::Sender<GatewayRequest>,
-    receiver: mpsc::Receiver<GatewayRequest>,
-) -> LnGateway {
-    let workdir = match plugin.option("fedimint-cfg") {
-        Some(options::Value::String(workdir)) => {
-            // FIXME: cln_plugin doesn't yet support optional parameters
-            if &workdir == "default-dont-use" {
-                panic!("fedimint-cfg option missing")
-            } else {
-                PathBuf::from(workdir)
-            }
-        }
-        _ => unreachable!(),
-    };
-    let host = match plugin.option("fedimint-host") {
-        Some(options::Value::String(host)) => host,
-        _ => unreachable!(),
-    };
-    let port = match plugin.option("fedimint-port") {
-        Some(options::Value::String(port)) => port,
-        _ => unreachable!(),
-    };
-    let bind_addr = format!("{}:{}", host, port)
-        .parse()
-        .expect("Invalid gateway bind address");
-
-    // If no config exists, try to generate one
-    let cfg_path = workdir.join("gateway.json");
-    let config = plugin.configuration();
-    let cln_rpc_socket = PathBuf::from(config.lightning_dir).join(config.rpc_file);
-    let mut ln_client = ClnRpc::new(cln_rpc_socket)
+    let config = rpc_config_mx
+        .lock()
         .await
-        .expect("connect to ln_socket");
-    if !Path::new(&cfg_path).is_file() {
-        generate_config(&workdir, &mut ln_client, &bind_addr).await;
-    }
-
-    // Run the gateway
-    let db_path = workdir.join("gateway.db");
-    let gw_client_cfg: GatewayClientConfig = load_from_file(&cfg_path);
-    let db = fedimint_rocksdb::RocksDb::open(db_path)
-        .expect("Error opening DB")
-        .into();
-    let ctx = secp256k1::Secp256k1::new();
-    let federation_client = Arc::new(Client::new(gw_client_cfg, db, ctx));
-    let ln_client = Arc::new(Mutex::new(ln_client));
+        .config()
+        .expect("Failed to get rpc config");
 
-    LnGateway::new(federation_client, ln_client, sender, receiver, bind_addr)
-}
+    let client = build_gateway_client(config.clone())?;
```

Comment:
Something is off with naming here. Why are we building "gateway client" (client *to call a gateway*) and then register it *in the gateway*)? So gateway can call itself?